CDCgov/prime-reportstream
# 7284: receiver checks for new transport

This PR adds support for the REST transport to the CheckFunction, so that the pickles-n-dogs status page can reflect the status of receivers using REST as well as SFTP. It also revises the functions in the CheckFunction class so that additional transports can be added.


Test Steps:
1. Pull down entire branch
2. Do a Gradle clean and package
3. Run dev environment
4. Ensure that there is a receiver configured for REST transport, in this example we use oh-doh
5. check the receiver using the command line with: curl -X POST -H "client:simple_report"  -d "" "http://localhost:7071/api/checkreceiver/org/oh-doh/receiver/elr"
6. confirm that the command returns JSON with "result": "success"
7. stop the REST mock container to simulate a disconnected receiver: docker-compose stop rest-webservice
8. check the receiver again with: curl -X POST -H "client:simple_report"  -d "" "http://localhost:7071/api/checkreceiver/org/oh-doh/receiver/elr"
9. confirm that the command returns JSON with "result": "fail"
10. Ensure that there is a receiver configured for SFTP transport, example ak-phd
11. check this receiver with: curl -X POST -H "client:simple_report"  -d "" "http://localhost:7071/api/checkreceiver/org/oh-doh/receiver/elr" 
12. confirm that the command returns JSON with "result": "success"
13. stop the SFTP mock container to simulate a disconnected receiver: docker-compose stop sftp
14. check this receiver with: curl -X POST -H "client:simple_report"  -d "" "http://localhost:7071/api/checkreceiver/org/oh-doh/receiver/elr"
15. confirm that the command returns JSON with "result": "fail"
16. run gradle testEnd2End
17. run integration test for RESTTransport
18. have a nice day


## Changes
- *revise CheckFunction to support REST check
- *refactor CheckFunction to constrain exception handling to the function calling the transport
- *refactor REST transport to allow call from check function
- *refactor REST transport to support logging when called from CheckFunction and/or called from Send function

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #7284
